src/rentACar/Application/Features/Brands/Rules/BrandBusinessRules.cs

```diff
@@ -16,17 +17,29 @@ public BrandBusinessRules(IBrandRepository brandRepository)
         _brandRepository = brandRepository;
     }
 
-    public async Task BrandIdShouldExistWhenSelected(int id)
+    public void BrandIdShouldExistWhenSelected(Brand? brand)
     {
-        Brand? result = await _brandRepository.GetAsync(b => b.Id == id);
-        if (result == null) throw new BusinessException(BrandMessages.BrandNotExists);
+        if (brand == null) throw new BusinessException(BrandMessages.BrandNotExists);
     }
 
     public async Task BrandNameCanNotBeDuplicatedWhenInserted(string name)
     {
-        IPaginate<Brand> result = await _brandRepository.GetListAsync(b => b.Name == name);
-        if (result.Items.Any()) throw new BusinessException(BrandMessages.BrandNameExists);
+        var result = await _brandRepository.Query().Where(x => x.Name == name).AnyAsync();
+        if (result) throw new BusinessException(BrandMessages.BrandNameExists);
+    }
+
+    public async Task BrandNameCanNotBeDuplicatedWhenUpdated(int id, string name)
+    {
+        var result = await _brandRepository.Query().Where(x => x.Name == name).AnyAsync();
```

Comment:
This section can be improved with a parameter change. Too many queries can be avoided.